File: dec01/node/part1.js
const fs = require('fs');
var os = require('os');

fs.readFile('../input.txt', 'utf-8', (err, data) => {
    let count = 0
    let old = null
    if (err) {
        console.error(err)
        return
    }
    data = data.split(os.EOL)

    data.forEach(element => {
        element = parseInt(element)
        if (old == null) {
            old = element
        } else {
            if (element > old) {
                count = count + 1
            }
            old = element
        }
    })
    console.log("There are", count, "depth increases")
})

File: dec01/c/part1.c
#include <stdio.h>
#include <stdlib.h>

typedef struct data
{
    int last;
    int depth_increase_count;
} data;

int main()
{
    FILE *fptr;
    data d = {0, 0};
    int current = 0;
    fptr = fopen("../input.txt", "rb");
    char line [1000];
    while(fgets(line, sizeof line, fptr) != NULL) {
        current = atoi(line);
        if (d.last == 0) {
            d.last = current;
            continue;
        } else {
            if (current > d.last) {
                d.depth_increase_count++;
            }
        }
        d.last = current;
    }
    fclose(fptr);
    printf("There are %d depth increases.\n", d.depth_increase_count);
    return 0;
}